addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/XSLTInjection.java

```diff
@@ -0,0 +1,274 @@
+        "No connection could be made because the target machine actively refused it",
+        "Can not load requested doc"
+    };
+
+    private static String[] cmdExecPayloads = {
+        "<xsl:variable name=\"rtobject\" select=\"runtime:getRuntime()\"/>\n"
+                + "<xsl:variable name=\"process\" select=\"runtime:exec($rtobject,'erroneous_command')\"/>\n"
+                + "<xsl:variable name=\"waiting\" select=\"process:waitFor($process)\"/>\n"
+                + "<xsl:value-of select=\"$process\"/>",
+        "<xsl:value-of select=\"php:function('exec','erroneous_command 2>&amp;1')\"/>"
+    };
+
+    private static String[] evidenceCmdExec = {
+        "Cannot run program", "erroneous_command: not found",
+    };
+
+    private static final Logger LOG = Logger.getLogger(XSLTInjection.class);
+
+    // used to check against the attack strength
+    private int requestsSent = 0;
+    // map setting the limits associated with each attack strength
+    private static Map<AttackStrength, Integer> strengthToRequestCountMap = new HashMap<>();
+
+    // Uses maps for checks to avoid having to pass
+    // the payloads, responses and resources strings as parameters
+    private static Map<XSLTInjectionCheck, String[]> payloads = new HashMap<>();
+    private static Map<XSLTInjectionCheck, String[]> responses = new HashMap<>();
+    private static Map<XSLTInjectionCheck, String> resourceIdentifiers = new HashMap<>();
+
+    static {
+        strengthToRequestCountMap.put(AttackStrength.DEFAULT, 12);
+        strengthToRequestCountMap.put(AttackStrength.LOW, 6);
+        strengthToRequestCountMap.put(AttackStrength.MEDIUM, 12);
+        strengthToRequestCountMap.put(AttackStrength.HIGH, 24);
+        strengthToRequestCountMap.put(AttackStrength.INSANE, 500);
+
+        payloads.put(XSLTInjectionCheck.ERROR, errorCausingPayloads);
+        responses.put(XSLTInjectionCheck.ERROR, evidenceError);
+        resourceIdentifiers.put(XSLTInjectionCheck.ERROR, "error");
+
+        payloads.put(XSLTInjectionCheck.VENDOR, vendorReturningPayloads);
+        responses.put(XSLTInjectionCheck.VENDOR, evidenceVendors);
+        resourceIdentifiers.put(XSLTInjectionCheck.VENDOR, "vendor");
+
+        responses.put(XSLTInjectionCheck.PORTSCAN, evidencePortScanning);
+        resourceIdentifiers.put(XSLTInjectionCheck.PORTSCAN, "portscan");
+
+        payloads.put(XSLTInjectionCheck.COMMAND_EXEC, cmdExecPayloads);
+        responses.put(XSLTInjectionCheck.COMMAND_EXEC, evidenceCmdExec);
+        resourceIdentifiers.put(XSLTInjectionCheck.COMMAND_EXEC, "command");
+    }
+
+    @Override
+    public void scan(HttpMessage msg, String param, String value) {
+        // initial check verifies if injecting certain strings causes XSLT related
+        // errors
+        if (tryInjection(msg, param, XSLTInjectionCheck.ERROR)) {
+            // verifies if we can get the vendor of the processor
+            tryInjection(msg, param, XSLTInjectionCheck.VENDOR);
```

Comment:
In case we have raised alert as injection is there , do we need to try for other injection types or should stop there ? As in general we stop if we find some alert. in your case you will still tryInjection for other types.

Reply:
read your discussion comment above. please ignore.